### PR TITLE
IKASAN-1594 - Add scheduled jobs to extract data from every module

### PR DESCRIPTION
**Pull request by andrzej-majewski**

### CODE REVIEW

File: ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            this.lastExecutionSuccessful = false;
+            throw new JobExecutionException("Could not execute housekeeping job: " + this.jobName, e);
+        }
+
+        this.lastExecutionSuccessful = true;
+        logger.debug("Finished harvesting job executing: " + this.getJobName());
+    }
+
+    @Override
+    public void save()
```
Comment (mitcje) [inline ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java]: Nothing to do here ?

File: ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    {
+        this.initialised = initialised;
+    }
+
+    @Override public Integer getThreadCount()
+    {
+        return null;
+    }
+
+    @Override public void setThreadCount(Integer threadCount)
```
Comment (mitcje) [inline ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java]: why empty implementation?

File: ikasaneip/spec/service/housekeeping/src/main/java/org/ikasan/spec/harvest/HarvestingSchedulerService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+public interface HarvestingSchedulerService
+{
+
+    void registerJobs();
+
+    void removeJob(String jobName);
+
+    void addJob(String jobName);
+
+    void startScheduler();
```
Comment (mitcje) [inline ikasaneip/spec/service/housekeeping/src/main/java/org/ikasan/spec/harvest/HarvestingSchedulerService.java]: Is this expected to have its own scheduler instance or use the shared scheduler of the module ?

Comment (andrzej-majewski) [reply ikasaneip/spec/service/housekeeping/src/main/java/org/ikasan/spec/harvest/HarvestingSchedulerService.java]: Well the startJob and stopJob method are not in use in context of a module. Bear in mind that HarvestingSchedulerService is used by both Dashboard and module, but the two are using different implementations of HarvestingJob. 

Hence startJob and stopJob methods are utilised by dashboard not a module. 

In context of a module we are using shared scheduler.

File: ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java (hunk 2 above)
Comment (andrzej-majewski) [reply ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java]: Above implementation is specific to the module and does not use multithreading. Dashboard Job implementation utilising solr is multithreaded.  One option would be to remove the method from the interface but that Dashboard code would need to be made aware of the implementation aka not coding agains interface.

File: ikasaneip/spec/service/housekeeping/src/main/java/org/ikasan/spec/harvest/HarvestingSchedulerService.java (hunk 3 above)
Comment (mitcje) [reply ikasaneip/spec/service/housekeeping/src/main/java/org/ikasan/spec/harvest/HarvestingSchedulerService.java]: maybe just stick a comment in the method to the effect of "not required"

File: ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java (hunk 2 above)
Comment (mitcje) [reply ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java]: maybe just stick a comment in the method to the effect of "not required"

File: ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java (hunk 1 above)
Comment (mitcje) [reply ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java]: any this one?

**Pull request merged**